### PR TITLE
Add Rumble Example

### PR DESCRIPTION
**Pull request by QuzarDC**
Adding an example for using the purupuru/JumpPack/rumble device. This is a recreation and enhancement of SinisterTengu's lost original which in turn was made into a tester for the vmu's beep by me, which was in turn added to examples by @gyrovorbis.

I've enhanced it over the original by allowing the controller or rumble to be disconnected and reconnected without having to restart the test (facilitating easier swapping and providing examples of such loops).

I've additionally updated the driver to de-duplicate the composed and raw versions of the rumble.

Before moving this out of draft, I may be adding a second mode which allows for a similar experimentation with the different composed effects elements.

### CODE REVIEW

**Comment by gyrovorbis** [comment]
Hell yeah! Much needed after the VMU beep one!

File: examples/dreamcast/rumble/rumble.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    pvr_init_defaults();
+
+    fnt = plx_font_load("/rd/axaxax.txf");
+    cxt = plx_fcxt_create(fnt, PVR_LIST_TR_POLY);
+
+    pvr_set_bg_color(0.0f, 0.0f, 0.0f);
+
+    /* Loop until Start is pressed */
+    for(;!(rel_buttons & CONT_START);) {
```
Comment (ljsebald) [inline examples/dreamcast/rumble/rumble.c:92]:
```suggestion
    while(!(rel_buttons & CONT_START)) {
```

File: examples/dreamcast/rumble/rumble.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        /* Start drawing the changable section of the screen */
+        w.x += 130; w.y += 120.0f;
+        plx_fcxt_setpos_pnt(cxt, &w);
+        plx_fcxt_setsize(cxt, 30.0f);
+        plx_fcxt_draw(cxt, "0x");
+
+        w.x += 48.0f;
+        plx_fcxt_setpos_pnt(cxt, &w);
+
+        for(count = 0; count <= 7; count++, w.x+=25.0f) {
```
Comment (ljsebald) [inline examples/dreamcast/rumble/rumble.c:119]:
```suggestion
        for(count = 0; count <= 7; count++, w.x += 25.0f) {
```

**Review by ljsebald — CHANGES_REQUESTED** [review]

File: doc/CHANGELOG [hunk 3]
```diff
@@ -174,6 +174,9 @@ KallistiOS version 2.1.0 -----------------------------------------------
       runtime aspects of the language [Andrew Apperley == AA]
 - DC  Improve fipr() and fipr_magnitude_sqr() functions [PC]
 - DC  Add optimized bit-reverse function && vmu_draw_lcd_rotated() [PC]
+- DC  Update bfont example for Dreamcast-specific characters [DH]
+- DC  Add example for VMU speaker use [DH] [FG]
```
Comment (ljsebald) [inline doc/CHANGELOG:178]:
```suggestion
- DC  Add example for VMU speaker use [DH && FG]
```

**Review by ljsebald — APPROVED** [review]

**Pull request merged**